src/document-context/measure-custom-elements.ts

```diff
@@ -0,0 +1,382 @@
+
+  customElements?: CustomElementsRegistry;
+}
+
+interface CustomElementsRegistry {
+  define(tagName: string, constructor: CustomElementV1Constructor):
+      CustomElementV1Constructor;
+}
+
+interface CustomElementV1Constructor {
+  new (): V1CustomElement;
+}
+
+interface V1CustomElement extends HTMLElement {
+  connectedCallback?(): void;
+  disconnectedCallback?(): void;
+  attributeChangedCallback?(): void;
+}
+
+interface ElementMeasurement {
+  tagName: string;
+  operation: string;
+  elementId: number|undefined;
+  duration: number;
+  start: number;
+  end: number;
+}
+
+type CallbackName = 'register' | 'created' | 'attached' | 'detached' |
+    'attributeChanged' | 'data';
+
+interface Console {
+  timeStamp(label: string): void;
+}
+
+(() => {
+  'use strict';
+
+  // No point without the measurement APIs we need (or polyfills thereof).
+  if (!window.performance || !window.performance.mark ||
+      !window.performance.measure) {
+    return;
+  }
+
+  const prefix = '🐰💫 ';
+
+  function makeMeasurement<R>(
+      operation: string, tagName: string, counter: number | null,
+      cb: () => R): R {
+    const counterSuffix = counter == null ? '' : ` ${counter}`;
+    const startMark = `${prefix}start ${operation} ${tagName}${counterSuffix}`;
+    const endMark = `${prefix}end ${operation} ${tagName}${counterSuffix}`;
+    const measure = `${prefix}${operation} ${tagName}${counterSuffix}`;
+
+    window.performance.mark(
+        `${prefix}start ${operation} ${tagName}${counterSuffix}`);
+    try {
+      return cb();
+    } finally {
+      window.performance.mark(endMark);
+      window.performance.measure(measure, startMark, endMark);
+    }
+  };
+
+  const idSymbol = Symbol('id');
+
+  const originalRegisterElement = document.registerElement;
+  if (originalRegisterElement) {
+    const boundRegisterElement: typeof originalRegisterElement =
+        originalRegisterElement.bind(document);
+    function wrappedRegisterElement(
+        tagName: string, options: RegisterElementOptions) {
+      const maybeProto = options.prototype;
+      if (!maybeProto) {
+        return boundRegisterElement(tagName, options);
+      }
+      const proto = maybeProto;
+      let elementCounter = 0;
+      const originalCreate = proto.createdCallback || (() => undefined);
+
+      function wrap(name: string, fullName: string) {
+        const original = proto[fullName] || (() => undefined);
+        proto[fullName] = function(this: any) {
+          const counter: number = this[idSymbol];
+
+          return makeMeasurement(
+              name, tagName, counter, () => original.apply(this, arguments));
+        };
+      }
+
+      proto.createdCallback = function(this: any) {
+        const counter = elementCounter++;
+        this[idSymbol] = counter
+
+        const result = makeMeasurement(
+            'created', tagName, counter, () => originalCreate.apply(this));
+
+        const originalPropertySetter = this._propertySetter;
```

Comment:
looks like yes! I guess this was an artifact of back when we were doing more patching of the Polymer() call?